Lib/test/test_typing.py

```diff
@@ -8478,6 +8478,22 @@ class TD2(TD1):
             b: str
 
         self.assertIs(TD2.__total__, True)
+        
+    def test_total_with_assigned_value(self):
+        class TD(TypedDict):
+            __total__ = "some_value"
+
+        self.assertIs(TD.__total__, True)
+
+        class TD2(TypedDict, total=True):
+            __total__ = "some_value"
+            
+        self.assertIs(TD2.__total__, True)
+
+        class TD3(TypedDict, total=False):
```

Comment:
On Python 3.12, I'm seeing this:

```py
>>> class A(typing.TypedDict, total="abc"):
...     __total__ = "def"
...
>>>
>>> A.__total__
'def'
```

Is it expected? if yes, please add a test, if not let's fix it/do something with it (cc @JelleZijlstra)

Reply:
This is no longer the case in 3.13+ where we always set `__total__` based on the class argument. I don't think your case is different from the ones already tested.

Reply:
Oops, right. I thought it was also in 3.12. My bad.